SourceX/miniwin/misc_msg.cpp

```diff
@@ -221,10 +290,160 @@ WINBOOL PeekMessageA(LPMSG lpMsg, HWND hWnd, UINT wMsgFilterMin, UINT wMsgFilter
 	}
 
 	lpMsg->hwnd = hWnd;
+	lpMsg->message = 0;
 	lpMsg->lParam = 0;
 	lpMsg->wParam = 0;
 
+	handle_touch(&e, MouseX, MouseY);
+	if (movie_playing) {
+		// allow plus button or mouse click to skip movie, no other input
+		switch (e.type) {
+		case SDL_CONTROLLERBUTTONDOWN:
+		case SDL_CONTROLLERBUTTONUP:
+			switch(e.cbutton.button) {
+			case SDL_CONTROLLER_BUTTON_START:
+			case SDL_CONTROLLER_BUTTON_A: // B on Switch
+				lpMsg->message = e.type == SDL_CONTROLLERBUTTONUP ? DVL_WM_LBUTTONUP : DVL_WM_LBUTTONDOWN;
+				lpMsg->lParam = (MouseY << 16) | (MouseX & 0xFFFF);
+				if (lpMsg->message == DVL_WM_LBUTTONUP)
+					lpMsg->wParam = keystate_for_mouse(0);
+				else
+					lpMsg->wParam = keystate_for_mouse(DVL_MK_LBUTTON);
+				break;
+			}
+			break;
+		case SDL_MOUSEBUTTONDOWN:
+		case SDL_MOUSEBUTTONUP:
+			if (e.button.button == SDL_BUTTON_LEFT) {
+				lpMsg->message = e.type == SDL_MOUSEBUTTONUP ? DVL_WM_LBUTTONUP : DVL_WM_LBUTTONDOWN;
+				lpMsg->lParam = (e.button.y << 16) | (e.button.x & 0xFFFF);
+				if (lpMsg->message == DVL_WM_LBUTTONUP)
+					lpMsg->wParam = keystate_for_mouse(0);
+				else
+					lpMsg->wParam = keystate_for_mouse(DVL_MK_LBUTTON);
+			}
+			break;
+		}
+		return true;
```

Comment:
This appears to be what breaks skipping videos via keyboard.

Reply:
fixed in https://github.com/diasurgical/devilutionX/pull/181/commits/0ac63d03b0b3a444f5333dcad864bed7fa93eb03